src/infrasys/time_series_manager.py

```diff
@@ -282,3 +286,26 @@ def _handle_read_only(self) -> None:
         if self._read_only:
             msg = "Cannot modify time series in read-only mode."
             raise ISOperationNotAllowed(msg)
+
+    def convert_storage(self, replace: bool = False, **kwargs) -> TimeSeriesStorageBase | None:
+        """
+        Create a new storage instance and copy all time series from the current to new storage
+
+        Parameters
+        ----------
+
+        replace: bool
+            if True, replace the current storage with the new storage, otherwise return the new storage
+
+        """
+        new_storage = self.create_new_storage(**kwargs)
+        for time_series_uuid in self.metadata_store.unique_uuids_by_type("SingleTimeSeries"):
```

Comment:
Can you use `SingleTimeSeries.__name__` instead? It may seem like a nit, but if we use the type and ever change the name, a refactor with the language server will automatically fix it.